### PR TITLE
new(tests): EIP-1014: Convert CREATE2 test from `ethereum/tests`

### PR DESCRIPTION
**Pull request by winsvega**
## 🗒️ Description
Port 
- src/GeneralStateTestsFiller/stCreate2/call_outsize_then_create2_successful_then_returndatasizeFiller.json
- src/GeneralStateTestsFiller/stCreate2/call_then_create2_successful_then_returndatasizeFiller.json

## 🔗 Related Issues
none

## ✅ Checklist
NOT READY FOR MERGE (NEED coverage)
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by winsvega** [comment]
@marioevz do you want the test coverage to be mandatory checkpoint as well? or perhaps we can setup a cli job to do it using evmone script that I was developing. hm..

**Comment by marioevz** [comment]
> @marioevz do you want the test coverage to be mandatory checkpoint as well? or perhaps we can setup a cli job to do it using evmone script that I was developing. hm..

I think we could start with a CLI entry point to check coverage, if you have an idea of implementation we could open another PR and follow it up there.

After that PR is merged we can do a follow up PR to try to embed it into the CI to automatically check it.

**Review by marioevz — COMMENTED** [review]
Looks great overall! Just some suggestions.

File: tests/constantinople/create2/test_create_returndata.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        address_to: Account(
+            balance=100000000,
+            nonce=0,
+            code=Op.JUMPDEST()
+            + Op.MSTORE(0x100, Op.CALLDATALOAD(0))
+            + Op.CALL(0x0900000000, address_call, 0, 0, 0, 0, call_return_size)
+            + Op.SSTORE(slot_returndatasize_before_create, Op.RETURNDATASIZE())
+            + make_create()
+            + Op.SSTORE(slot_returndatasize_after_create, Op.RETURNDATASIZE())
+            + Op.SSTORE(slot_code_worked, 1)
```
[the remark this answers is not in the document]
Comment (winsvega) [reply tests/constantinople/create2/test_create_returndata.py:67]: hm, the values does not match when returnsize is not 32. need to debug

File: .github/workflows/coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+name: Evmone Coverage Report
```
Comment (marioevz) [inline .github/workflows/coverage.yaml:1]: I have not reviewed this file yet, but I think we need to split this file into its own PR to have it properly reviewed, since it's completely orthogonal to test implementation.

Comment (winsvega) [reply .github/workflows/coverage.yaml:1]: Yes, but on this pr, I can test directly that it works with the given test changes.

It's not ready yet.

File: tests/constantinople/create2/test_create_returndata.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
+"""
```
Comment (marioevz) [inline tests/constantinople/create2/test_create_returndata.py:1]: Could we take this opportunity to rename folder `create2` to ` eip1014_create2`?

Comment (winsvega) [reply tests/constantinople/create2/test_create_returndata.py:1]: Yes, also need to enable coverage

File: tests/constantinople/eip1014_create2/test_create_returndata.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    # Storage vars
+    slot_returndatasize_before_create = 0
+    slot_returndatasize_after_create = 1
+    slot_return_data_hash_before_create = 2
+    slot_return_data_hash_after_create = 3
+    slot_code_worked = 4
+
+    # CREATE2 Initcode
+    create2_salt = 1
+    initcode = Op.MSTORE(0, 0x112233) + Op.RETURN(0, 32) + Op.STOP()
```
Comment (marioevz) [inline tests/constantinople/eip1014_create2/test_create_returndata.py:41]:
```suggestion
    initcode = Op.MSTORE(0, 0x112233) + Op.RETURN(0, 32)
```
`Op.STOP` is unnecessary here.

File: tests/constantinople/eip1014_create2/test_create_returndata.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                slot_return_data_hash_before_create: keccak256(call_return_data),
+                slot_return_data_hash_after_create: keccak256(empty_data),
+            }
+        )
+    }
+
+    tx = Transaction(
+        ty=0x0,
+        chain_id=0x0,
+        sender=pre.fund_eoa(7_000_000_000_000_000_000),
```
Comment (marioevz) [inline tests/constantinople/eip1014_create2/test_create_returndata.py:97]:
```suggestion
        sender=pre.fund_eoa(),
```
We can just use the default in this case.

File: tests/constantinople/eip1014_create2/test_create_returndata.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                slot_returndatasize_after_create: 0,
+                slot_return_data_hash_before_create: keccak256(call_return_data),
+                slot_return_data_hash_after_create: keccak256(empty_data),
+            }
+        )
+    }
+
+    tx = Transaction(
+        ty=0x0,
+        chain_id=0x0,
```
Comment (marioevz) [inline tests/constantinople/eip1014_create2/test_create_returndata.py:96]:
```suggestion
```
These two are unnecessary.

File: tests/constantinople/eip1014_create2/test_create_returndata.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        sender=pre.fund_eoa(7_000_000_000_000_000_000),
+        to=address_to,
+        gas_price=10,
```
Comment (marioevz) [inline tests/constantinople/eip1014_create2/test_create_returndata.py:99]:
```suggestion
```
Also unnecessary.

**Review by marioevz — COMMENTED** [review]
Some comments, thanks!

File: tests/constantinople/eip1014_create2/test_create_returndata.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    address_to = pre.deploy_contract(
+        balance=100_000_000,
+        code=Op.JUMPDEST()
+        + Op.MSTORE(0x100, Op.CALLDATALOAD(0))
+        + Op.CALL(0x0900000000, address_call, 0, 0, 0, 0, call_return_size)
+        + Op.SSTORE(slot_returndatasize_before_create, Op.RETURNDATASIZE())
+        + Op.SSTORE(slot_return_data_hash_before_create, Op.SHA3(0, call_return_size))
+        + make_create()
+        + Op.SSTORE(slot_returndatasize_after_create, Op.RETURNDATASIZE())
+        + Op.SSTORE(slot_return_data_hash_after_create, Op.SHA3(0, Op.RETURNDATASIZE()))
```
[the remark this answers is not in the document]
Comment (winsvega) [reply tests/constantinople/eip1014_create2/test_create_returndata.py:71]: if I call returndatacopy with the size that is > returndatasize it makes evm failure

[the remark this answers is not in the document]
Comment (winsvega) [reply tests/constantinople/eip1014_create2/test_create_returndata.py]: added more parametrization

**Review by marioevz — APPROVED** [review]
LGTM, thanks!
